zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -94,8 +94,12 @@ where
     /// [`ShieldedData`] validates this [value balance consensus
     /// rule](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus):
     ///
-    /// "If effectiveVersion = 4 and there are no Spend descriptions or Output
-    /// descriptions, then valueBalanceSapling MUST be 0."
+    /// # Consensus
+    ///
+    /// > [Sapling onward] If effectiveVersion = 4 and there are no Spend
+    /// > descriptions or Output descriptions, then valueBalanceSapling MUST be 0.
+    ///
+    /// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
```

Comment:
I couldn't find where we enforce this :thinking: 
Can you please double-check? If we don't we should create an issue for it

Reply:
It seems to be explained here: https://github.com/ZcashFoundation/zebra/blob/4089e5a73e0637f6a31761a876f69f69d14a3098/zebra-chain/src/sapling/shielded_data.rs#L107-L111

Reply:
Ok, i follow you now and it seems you are right. I was about to add code into this PR for it but i think it will be better to do it separated. Thanks for the find!

Reply:
https://github.com/ZcashFoundation/zebra/issues/3478

Reply:
The transaction v4 sapling value balance check is here:
https://github.com/ZcashFoundation/zebra/blob/1a14baf8e7c6c82dba7a64e9b43d076b8f856f2a/zebra-chain/src/transaction/serialize.rs#L658-L663

This is v5 sapling deserialization, which doesn't need any checks, because the value balance isn't encoded when there are no spends and no outputs:
https://github.com/ZcashFoundation/zebra/blob/a1f4cec0de599516d8693e29f3e81b00db9e634e/zebra-chain/src/transaction/serialize.rs#L191-L198

Reply:
Ok, thats good. thanks for the explication.

Reply:
Makes sense, i did this change at https://github.com/ZcashFoundation/zebra/pull/3461/commits/6fbdc1369cff14f6545c7bfa290bfc851b2cc7db, also removed almost all the description of the field as i think it was more confusing than anything else.